lib/libc/minimal/include/sys/types.h

```diff
@@ -32,27 +32,8 @@ typedef __SIZE_TYPE__ ssize_t;
 
 #if !defined(__off_t_defined)
 #define __off_t_defined
-
-#if defined(__i386) || defined(__x86_64) || defined(__ARC64__)
-typedef long int off_t; /* "long" works for all of i386, X32 and true 64 bit */
-#elif defined(__ARM_ARCH)
-typedef int off_t;
-#elif defined(__arc__)
-typedef int off_t;
-#elif defined(__NIOS2__)
-typedef int off_t;
-#elif defined(__riscv)
-typedef int off_t;
-#elif defined(__XTENSA__)
-typedef int off_t;
-#elif defined(__sparc__)
-typedef int off_t;
-#elif defined(__mips)
-typedef int off_t;
-#else
-#error "The minimal libc library does not recognize the architecture!\n"
-#endif
-
+/* off_t is defined such that it matches the size of a pointer */
+typedef __INTPTR_TYPE__ off_t;
```

Comment:
`off_t` is not designated to be a pointer https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/sys_types.h.html

Reply:
Have you read the commit message?

> While the POSIX standard does not specify the size requirement of the
off_t type, it generally corresponds to the size of a pointer in
practice, mainly because the optimal file handling size is closely tied
to the native pointer size.

Reply:
```
 /* off_t is defined such that it matches the size of a pointer */
```
The comment describes what happens next line, which the line explains itself, but not why.

> Have you read the commit message?

Yes I have, still 
> 
> > (...)  it [off_t type]  generally corresponds to the size of a pointer in
> > practice, mainly because the optimal file handling size is closely tied
> > to the native pointer size.

did not convince me. I just do not see that happening to often.

By optimal you mean from the point of arithmetic right? Because we assume that pointer is processor word size and that arithmetic is fastest this way.
And does it really happen so often in practice? As far as I can tell apps access files through buffers, they may index them or use pointer arithmetic on them, but still this is not direct usage of file offset with memory objects.

The `off_t` is not a pointer, it is an offset, and should not be used as offset to some memory object pointer;  I believe it should be defined as an integer type (`long`), for all arch.

I have checked FreeBSD VM subsy, just for reference, and it does seem to use `off_t` mainly as offset within files (devices); it uses different pointer types for objects that relate to file system, but `off_t` values are never directly assigned to them; and `off_t` things does not seem to be used to index memory objects.

Further or later we will have to make this configurable between `long` and `long long` anyway as somebody will finally connect 8GB flash and will ask why it rolls over at 4GB.

Reply:
> (...) I do not really see how this "pointer" vs "integer" argument makes sense in this case.

That was me being tired, and missing some words while reading description.

> In this case, intptr_t is as much of a good answer as long and vice versa. It is just that, in my opinion, intptr_t is a marginally better choice than discrete int or long because it is less "arbitrarily defined."

Touché, @stephanosio.